assembly/__tests__/i128.spec.ts

```diff
@@ -90,6 +91,111 @@ describe("Basic Operations", () => {
+  });
+
+  it("Should invariant left shift zero number", () => {
+    var a = new i128(1, 1);
+    expect(a << 0).toStrictEqual(a);
+  });
+
+  it("Should right shift one number", () => {
+    var a = new i128(0, 100);
+    expect(a >> 65).toStrictEqual(new i128(50));
+  });
+  it("Should right shift for negative number", () => {
+    var a = i128.from(-64);
+    expect(a >> 2).toStrictEqual(i128.from(-16));
+  });
+
+  it("Should right shift for negative number 1", () => {
+    var a = i128.from(-10000000);
+    expect(a >> 16).toStrictEqual(i128.from(-152));
+  });
+
+  it("Should right shift for negative number 2", () => {
+    var a = i128.from(-100000000000);
+    expect(a >> 32).toStrictEqual(i128.from(-23));
+  });
+
+  it("Should right shift for negative number 3", () => {
+    var a = i128.fromString("-922337203685477580800");
+    expect(a >> 63).toStrictEqual(i128.from(-100));
+  });
+
+  it("Should right shift for negative number 4", () => {
+    var a = i128.fromString("-922337203685477580800");
+    expect(a >> 64).toStrictEqual(i128.from(-50));
+  });
+
+  it("Should right shift for negative number 5", () => {
+    var a = i128.fromString("-170141183460469231731687303715884105728");
+    expect(a >> 127).toStrictEqual(i128.from(-1));
+  });
+
+  it("Should periodic right shift one number", () => {
+    var a = new i128(0, 100);
+    expect(a >> (65 + 128)).toStrictEqual(new i128(50));
+  });
+
+  it("Should invariant right shift zero number", () => {
+    var a = new i128(1, 1);
+    expect(a >> 0).toStrictEqual(a);
+  });
+
+  it("Should multiplication 1", () => {
+    let a = u128.from(57326);
+    let b = u128.from(144873);
+    expect(i128.mul(a.toI128(),b.toI128())).toStrictEqual(u128.mul(a,b).toI128());
+  });
+  it("Should multiplication 2", () => {
+    let a = i128.One;
+    expect(i128.mul(a,a)).toStrictEqual(a);
+  });
+  it("Should multiplication 3", () => {
+    let a = i128.One;
+    let b = i128.fromI32(2);
+    expect(i128.mul(a,b)).toStrictEqual(b);
+  });
+  it("Should multiplication 4", () => {
+    let b = i128.fromI32(2);
+    let c = i128.fromI32(3);
+    let d = i128.fromI32(6);
+    expect(i128.mul(b,c)).toStrictEqual(d);
+  });
+  it("Should Division 1", () => {
+    let a = u128.from(7);
+    let b = u128.from(144873);
+    expect(i128.div(b.toI128(),a.toI128())).toStrictEqual(u128.div(b,a).toI128());
+  });
+  it("Should Division 2", () => {
+    let a = i128.One;
+    expect(i128.div(a,a)).toStrictEqual(a);
+  });
+  it("Should Division 3", () => {
+    let a = i128.One;
+    let b = i128.fromI32(2);
+    expect(i128.div(b,a)).toStrictEqual(b);
+  });
+  it("Should Division 4", () => {
+    let b = i128.fromI32(2);
+    let c = i128.fromI32(3);
+    let d = i128.fromI32(6);
+    expect(i128.div(d,c)).toStrictEqual(b);
+  });
+  it("Should Division 5", () => {
+    let b = i128.fromI32(2);
+    let c = i128.fromI32(3);
+    let d = i128.fromI32(6);
+    expect(i128.div(d,b)).toStrictEqual(c);
```

Comment:
these two tests are almost identical and don't really cover anything. It would be nice to have these cases: -1 / 3, -5 / 3, -7 / -2, 7 / -3 and `min / -1` (which should throw error)